src/components/Views/CameraView/CameraView.js

```diff
@@ -101,87 +129,239 @@ class CameraView extends React.Component<
+              <TouchableOpacity
+                style={styles.cancelButton}
+                onPress={() => this.props.navigation.goBack()}
+              >
+                <Text style={styles.cancelText}>Cancel</Text>
+              </TouchableOpacity>
+            )}
+          </RNCamera>
+          {loading && (
+            <View
               style={{
-                width: 125,
-                height: 125
+                position: 'absolute',
+                height: fromEditor
+                  ? Dimensions.get('window').height - 30
+                  : Dimensions.get('window').height,
+                width: Dimensions.get('window').width,
+                backgroundColor: 'rgba(66,66,66,.8)',
+                alignItems: 'center',
+                justifyContent: 'center'
               }}
-            />
-          </TouchableOpacity>
-          <TouchableOpacity
-            style={styles.cancelButton}
-            onPress={() => this.props.navigation.goBack()}
-          >
-            <Text style={styles.cancelText}>Cancel</Text>
-          </TouchableOpacity>
-        </RNCamera>
-        {loading && (
-          <View
-            style={{
-              position: 'absolute',
-              height: Dimensions.get('window').height - 30,
-              width: Dimensions.get('window').width,
-              backgroundColor: 'rgba(66,66,66,.8)',
-              alignItems: 'center',
-              justifyContent: 'center'
+            >
+              <View
+                style={{
+                  backgroundColor: '#FFF',
+                  paddingHorizontal: 50,
+                  paddingVertical: 30,
+                  justifyContent: 'center',
+                  borderRadius: 3
+                }}
+              >
+                <ActivityIndicator />
+                <Text>{I18n.t('saving_image')}</Text>
+              </View>
+            </View>
+          )}
+        </View>
+      );
+    }
+    return (
+      <Drawer
+        ref={this.handleRightDrawerRef}
+        content={<ObservationsView closeRightDrawer={this.closeRightDrawer} />}
+        onCloseStart={onDrawerClose}
+        onOpenStart={onDrawerOpen}
+        openDrawerOffset={0}
+        side="right"
+        type="displace"
+      >
+        <Header
+          leftIcon={
+            <TouchableOpacity onPress={this.goToMapView}>
+              <Icon color={WHITE} name="map" size={30} />
+            </TouchableOpacity>
+          }
+          rightIcon={
+            <TouchableOpacity onPress={this.openRightDrawer}>
+              <CollectionsImg color={WHITE} name="collections" size={30} />
+            </TouchableOpacity>
+          }
+          style={{
+            position: 'absolute',
+            top: 0,
+            left: 0,
+            right: 0,
+            zIndex: 5
+          }}
+        />
+        {showSavedModal && <SavedModal />}
+        <View style={{ flex: 1, flexDirection: 'column' }}>
+          <RNCamera
```

Comment:
Seems like some of this code is redundant. You might want to put it in a separate component or function `renderCamera`

Reply:
Ya will refactor a bit